Mathlib/CategoryTheory/Generator/Presheaf.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import Mathlib.CategoryTheory.Generator.Coproduct
+import Mathlib.CategoryTheory.Limits.FunctorCategory.Basic
+
+/-!
+# Generators in the category of presheaves
+
+In this file, we show that if `A` is a category with zero morphisms that
+has a separator (and suitable coproducts), then the category of
+presheaves `Cᵒᵖ ⥤ A` also has a separator.
+
+-/
+
+universe w v' v u' u
+
+namespace CategoryTheory
+
+open Limits Opposite
+
+namespace Presheaf
+
+variable {C : Type u} [Category.{v} C] {A : Type u'} [Category.{v'} A]
+  [HasCoproducts.{v} A]
+
+/-- Given `X : C` and `M : A`, this is the presheaf `Cᵒᵖ ⥤ A` which sends
+`Y : Cᵒᵖ` to the coproduct of copies of `M` indexed by `Y.unop ⟶ X`. -/
+@[simps]
+noncomputable def freeYoneda (X : C) (M : A) : Cᵒᵖ ⥤ A where
+  obj Y := ∐ (fun (i : (yoneda.obj X).obj Y) ↦ M)
```

Comment:
Hmm, this makes me feel like we should have abbreviations for powers and copowers... What do you think? (If you agree, it doesn't need to be in this PR though!)

Reply:
I am not completely sure we need this, as it is quite rarely used.